enterprise/server/remote_execution/dirtools/dirtools.go

```diff
@@ -680,16 +718,27 @@ func (ff *BatchFileFetcher) FetchFiles(filesToFetch FileMap, opts *DownloadTreeO
 		}
 		return r
 	}
-	req := newRequest()
 
-	currentBatchRequestSize := int64(0)
-	eg, ctx := errgroup.WithContext(ff.ctx)
+	linkEG, _ := errgroup.WithContext(ff.ctx)
+	if runtime.GOOS == "darwin" {
+		// Macs don't appear to like too much parallelism, with overall
+		// performance dropping with high parallelism.
+		linkEG.SetLimit(5)
+	} else {
+		linkEG.SetLimit(100)
```

Comment:
does it make sense to max out at runtime.GOMAXPROCS(0)?

Reply:
done